### PR TITLE
Split tasks off the user show page

### PR DESCRIPTION
**Pull request by hennevogel**
Account info and involved projects/packages are fairly static info. All the
things collaboration, requests, review and maintenance updates are ever
changing. Give each of them a dedicated page to provide better focus.

screenshot

### CODE REVIEW

File: src/api/app/controllers/webui/user_controller.rb [hunk 1]
```diff
@@ -54,9 +54,10 @@ def show
     @owned = @displayed_user.owned_packages
     @groups = @displayed_user.groups
     @role_titles = @displayed_user.roles.global.pluck(:title)
+  end
 
-    return unless User.current == @displayed_user
-    @is_displayed_user = User.current == @displayed_user
+  def tasks
```
Comment (ChrisBr) [inline src/api/app/controllers/webui/user_controller.rb]: Can we move that to a dedicated controller, e.g. ``Webui::User::TasksController`` and then create a show / index action?

Comment (hennevogel) [reply src/api/app/controllers/webui/user_controller.rb]: @ChrisBr done

File: src/api/app/views/webui/user/tasks.html.haml [hunk 2]
```diff
@@ -0,0 +1,79 @@
+- @pagetitle = "Tasks for #{@displayed_user}"
```
Comment (ChrisBr) [inline src/api/app/views/webui/user/tasks.html.haml]: This one is not necessary anymore, right?

File: src/api/config/routes.rb [hunk 3]
```diff
@@ -358,6 +358,8 @@ def self.public_or_about_path?(request)
     get 'user/notifications' => :index, constraints: cons, controller: 'webui/users/subscriptions'
     put 'user/notifications' => :update, constraints: cons, controller: 'webui/users/subscriptions'
 
+    get 'user/taks/:user' => :index, constraints: cons, controller: 'webui/users/tasks', as: 'user_tasks'
```
Comment (ChrisBr) [inline src/api/config/routes.rb]: ``'user/tasks/:user'``

Comment (ChrisBr) [reply src/api/config/routes.rb]: What about 
```
get 'home/my_work' => :home
get 'home/list_my' => :home
```

File: src/api/app/controllers/webui/users/tasks_controller.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+module Webui
+  module Users
+    class TasksController < WebuiController
+      before_action :require_login
+      before_action :check_display_user
+
+      def index
+        redirect_back(fallback_location: root_path, notice: 'Sorry you\'re not allowed to view tasks of other users') unless @is_displayed_user
```
Comment (ChrisBr) [inline src/api/app/controllers/webui/users/tasks_controller.rb]: I'm wondering if we even need to do that?
Why not just use ``User.current``? And the route would also be easier ``/user/tasks``, no need to add the login as a parameter.

File: src/api/config/routes.rb (hunk 3 above)
Comment (ChrisBr) [reply src/api/config/routes.rb]: See my other comment, why not dropping the parameter and just use ```User.current```?

File: src/api/app/views/webui/user/tasks.html.haml (hunk 2 above)
Comment (hennevogel) [reply src/api/app/views/webui/user/tasks.html.haml]: It is. Makes this:

screenshot from 2018-03-14 14-08-39

File: src/api/app/controllers/webui/webui_controller.rb [hunk 5]
```diff
@@ -207,6 +207,7 @@ def check_display_user
       @displayed_user = User.current
       @displayed_user ||= User.find_nobody!
     end
+    @is_displayed_user = User.current == @displayed_user
```
[the remark this answers is not in the document]
Comment (hennevogel) [reply src/api/app/controllers/webui/webui_controller.rb]: it's used in `#show`

**Pull request merged**